tensorboard/writer/record_writer.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""
+To write tf_record into file. Here we use it for tensorboard's event writting.
+The code was borrowed from https://github.com/TeamHG-Memex/tensorboard_logger
+"""
+
+import copy
+import io
+import os.path
+import re
+import struct
+try:
+    import boto3
+    S3_ENABLED = True
+except ImportError:
+    S3_ENABLED = False
+
+from .crc32c import crc32c
+
+
+_VALID_OP_NAME_START = re.compile('^[A-Za-z0-9.]')
+_VALID_OP_NAME_PART = re.compile('[A-Za-z0-9_.\\-/]+')
+
+# Registry of writer factories by prefix backends.
+#
+# Currently supports "s3://" URLs for S3 based on boto and falls
+# back to local filesystem.
+REGISTERED_FACTORIES = {}
+
+
+def register_writer_factory(prefix, factory):
+    if ':' in prefix:
+        raise ValueError('prefix cannot contain a :')
+    REGISTERED_FACTORIES[prefix] = factory
+
+
+def directory_check(path):
+    '''Initialize the directory for log files.'''
+    try:
+        prefix = path.split(':')[0]
+        factory = REGISTERED_FACTORIES[prefix]
+        return factory.directory_check(path)
+    except KeyError:
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+
+def open_file(path):
+    '''Open a writer for outputting event files.'''
+    try:
+        prefix = path.split(':')[0]
+        factory = REGISTERED_FACTORIES[prefix]
+        return factory.open(path)
+    except KeyError:
+        return open(path, 'wb')
+
+
+class S3RecordWriter(object):
```

Comment:
Let's remove S3 support for now and we'll add that by extending gfile in the compat layer in a later PR.